app/Main.hs

```diff
@@ -4,18 +4,15 @@ import           TwitchAPI          (fetchViewerCount)
 
 import           Control.Concurrent (threadDelay)
 import           System.Environment (getArgs)
+import           Control.Monad (forever)
 
 main :: IO ()
 main = do
   args <- getArgs
-  watch 10 (process (getOutputPath args))
-
-watch :: Int -> IO () -> IO ()
-watch seconds action = do
-  action
-  putStrLn "."
-  threadDelay (toMicroseconds seconds)
-  watch seconds action
+  forever $ process (getOutputPath args) >> putStrLn "." >> wait 10
```

Comment:
I would rather have kept the do block and wrap it with forever, something like this maybe:

```haskell
main :: IO ()
main = do
  args <- getArgs
  forever $ do
    process (getOutputPath args)
    putStrLn "."
    wait 10
```

Reply:
You are right, thanks for your advice.